examples/recorder-crx/src/background.ts

```diff
@@ -73,57 +78,65 @@ async function changeAction(tabId: number, mode?: CrxMode | 'detached') {
 // https://bugs.chromium.org/p/chromium/issues/detail?id=1450904
 chrome.tabs.onUpdated.addListener(tabId => changeAction(tabId));
 
-async function getCrxApp(isIncognito: boolean) {
-  if (!crxAppPromise) {
-    await settingsInitializing;
-
-    crxAppPromise = crx.start( { incognito: isIncognito } ).then(crxApp => {
-      crxApp.recorder.addListener('hide', async () => {
-        await crxApp.detachAll();
-      });
-      crxApp.recorder.addListener('modechanged', async ({ mode }) => {
-        await Promise.all([...attachedTabIds].map(tabId => changeAction(tabId, mode)));
-      });
-      crxApp.addListener('attached', async ({ tabId }) => {
-        attachedTabIds.add(tabId);
-        await changeAction(tabId, crxApp.recorder.mode());
-      });
-      crxApp.addListener('detached', async tabId => {
-        attachedTabIds.delete(tabId);
-        await changeAction(tabId, 'detached');
-      });
-      setTestIdAttributeName(settings.testIdAttributeName);
+async function getCrxApp(incognito: boolean) {
+  // always start a new crx app.
```

Comment:
I would keep the previous code, and only start `crxApp` when `crxAppPromise` is undefined (hiding recorder will close it and set it as undefined)